hessian-lite/src/main/java/com/alibaba/com/caucho/hessian/io/MapDeserializer.java

```diff
@@ -163,13 +163,11 @@ public Object readObject(AbstractHessianInput in,
         return map;
     }
 
-    private Map createMap()
+    private Map createMap(int expectedSize)
             throws IOException {
 
-        if (_type == null)
-            return new HashMap();
-        else if (_type.equals(Map.class))
-            return new HashMap();
+        if (_type == null || _type.equals(Map.class) || _type.equals(HashMap.class))
```

Comment:
could you provide a test case that _type could be HashMap?

Reply:
i see, it could be tested through generic testing if the server returns a object which class that client does not have.